File: Sources/windows/yage/capabilities/rotating.py
from yage.models.capability import Capability

class Rotating(Capability):
    def __init__(self, subject):
        super().__init__(subject)
        self.is_enabled = False
        self.z_angle = 0
        self.is_flipped_vertically = False
        self.is_flipped_horizontally = False
        self.subject.rotation = self

File: Sources/macOS/Sources/Yage/Capabilities/Rotating.swift
import SwiftUI

public class Rotating: Capability {
    public var zAngle: CGFloat = 0
    public var isFlippedVertically: Bool = false
    public var isFlippedHorizontally: Bool = false
    
    public override func install(on subject: Entity) {
        super.install(on: subject)
        isEnabled = false
    }
}

public extension Entity {
    var rotation: Rotating? {
        capability(for: Rotating.self)
    }
}
